### PR TITLE
Events creation and edit bug fix

### PR DESCRIPTION
**Pull request by abdullai-t**
####  Summary / Highlights
This PR contains code that validates start and end date and time of event during creation and edit. Related [issue](https://github.com/massenergize/frontend-admin/issues/716)
#### Requirements (place an `x` in each `[ ]`)
* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [ ] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

### CODE REVIEW

File: src/api/store/event.py [hunk 1]
```diff
@@ -225,6 +225,8 @@ def create_event(self, context: Context, args) -> Tuple[dict, MassEnergizeAPIErr
       day_of_week = args.pop('day_of_week', None)
       week_of_month = args.pop("week_of_month", None)
       final_date = args.pop('final_date', None)
+      if end_date_and_time < start_date_and_time and not is_recurring :
```
Comment (Opoku-Agyemang) [inline src/api/store/event.py:228]: why are we checking is_recurring here?

**Review by Opoku-Agyemang — CHANGES_REQUESTED** [review]
can we add a test for this?

**Comment by Davidjwkinghorn** [comment]
> can we add a test for this?

I agree, I'll make a separate ticket for that

**Review by Davidjwkinghorn — APPROVED** [review]
LGTM

**Pull request merged**